src/Sarif.Multitool/Rules/SARIF2009.UseConventionalSymbolicNames.cs

```diff
@@ -0,0 +1,105 @@
+            }
+        }
+
+        protected override void Analyze(Tool tool, string toolPointer)
+        {
+            if (tool.Driver != null)
+            {
+                AnalyzeToolDriver(tool.Driver, toolPointer.AtProperty(SarifPropertyName.Driver));
+            }
+        }
+
+        private void AnalyzeToolDriver(ToolComponent toolComponent, string toolDriverPointer)
+        {
+            if (toolComponent.Rules != null)
+            {
+                string rulesPointer = toolDriverPointer.AtProperty(SarifPropertyName.Rules);
+                for (int i = 0; i < toolComponent.Rules.Count; i++)
+                {
+                    AnalyzeReportingDescriptor(toolComponent.Rules[i], rulesPointer.AtIndex(i));
+                }
+            }
+        }
+
+        private void AnalyzeReportingDescriptor(ReportingDescriptor reportingDescriptor, string reportingDescriptorPointer)
+        {
+            if (string.IsNullOrWhiteSpace(reportingDescriptor.Id))
+            {
+                return;
+            }
+
+            if (!s_conventionalIdRegex.IsMatch(reportingDescriptor.Id))
+            {
+                // {0}: The 'id' property of the rule '{1}' does not follow the recommended format:
+                // a short string identifying the tool concatenated with a numeric rule number, for example, `CS2001`.
+                // Using a conventional format for the rule id provides a more uniform experience across tools.
+                LogResult(
+                    reportingDescriptorPointer.AtProperty(SarifPropertyName.Id),
+                    nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_IncludeDynamicContent_Text),
+                    reportingDescriptor.Name);
+            }
+        }
+
+        private void AnalyzeOriginalUriBaseIds(IDictionary<string, ArtifactLocation> originalUriBaseIds, string originalUriBaseIdsPointer)
+        {
+            foreach (KeyValuePair<string, ArtifactLocation> originalUriBaseId in originalUriBaseIds)
+            {
+                if (!s_conventionalSymbols.Contains(originalUriBaseId.Key))
+                {
+                    // {0}: The 'originalUriBaseIds' symbol '{1}' is not one of the conventional symbols. 
+                    // We suggest 'REPOROOT' for the root of a repository, 'SRCROOT' for the root of the directory containing all source code,
+                    // 'TESTROOT' for the root of the directory containing all test code (if your repository is organized in that way),
+                    // and 'BINROOT' for the root of the directory containing build output (if your project places all build output in a common directory).
+                    LogResult(
+                        originalUriBaseIdsPointer.AtProperty(originalUriBaseId.Key),
+                        nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_IncludeDynamicContent_Text),
```

Comment:
>SARIF2001_AuthorHighQualityMessages_Warning_IncludeDynamicContent_Text [](start = 45, length = 70)

resource name is wrong?

Reply:
just updated! thanks!

---
In reply to: [445549288](https://github.com/microsoft/sarif-sdk/pull/1938#discussion_r445549288) [](ancestors = 445549288)